app/Http/Controllers/TechtreeController.php

```diff
@@ -297,4 +298,38 @@ public function getPropertyTooltip(string $name, array $breakdown, int $value):
             'property_value' => $value,
         ]);
     }
+    /**
+     * Returns techtree plasma table.
+     *
+     * @param GameObject $object
+     * @param PlayerService $player
+     * @return View
+     * @throws Exception
+     */
+    public function getPlasmaTable(GameObject $object, PlayerService $player): View
+    {
+        if ($object->type != 'research') {
+            return view('empty');
+        }
+        
+        $planet = $player->planets->current();
+        $current_level = $player->planets->current()->getObjectLevel($object->machine_name);
+        $plasma_table = [];
+        $min_level = (($current_level - 2) > 1) ? $current_level - 2 : 1;
+        for ($i = $min_level; $i < $min_level + 15; $i++) {
+
+            $plasma_table[] = [
+                'level' => $i,
+                'metal_bonus' => $i,
+                'crystal_bonus' => $i * 0.66,
+                'deuterium_bonus' => $i * 0.33
```

Comment:
😻  Looks good! 

ℹ️ However for documentation purposes: later when implementing these bonuses into the actual resource production of the planet these values will probably be need to moved to a central location.

Currently objects like "metal mine" have the following formula defined in the BuildingObjects.php file to calculate the production value:
```
        $metalMine->production = new GameObjectProduction();
        $metalMine->production->metal = 'return (30 * $building_level * pow((1.1), $building_level)) * (0.1 * $building_percentage);';
        $metalMine->production->energy = 'return - (10 * $building_level * pow((1.1), $building_level)) * (0.1 * $building_percentage);';
```

These formulas are ultimately evaluated by the `PlanetService::getBuildingProduction()` method.

This production values system currently only works for buildings on a planet.  So we'll have to define something similarly for these tech bonuses so the bonuses on top of the regular production can be calculated.

If you want to feel free to take a look at this, but IMO it's not a blocker for this issue and I'm happy to merge as-is. We can add a new issue to the backlog for the future. 😄 .

Reply:
I understood it slightly 😆, as I've never played this game, but I can get it later from the backlog when I know better 👍.